### PR TITLE
chore: Migrate to pnpm

### PR DESCRIPTION
**Pull request by nahsi**

### CODE REVIEW

File: ci.cjs [hunk 1]
```diff
@@ -0,0 +1,160 @@
+#! /usr/bin/env node
```
Comment (folex) [inline ci.cjs:1]: Maybe put it to .github?

File: ci.cjs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      postfix = process.argv[3];
+      if (!postfix) {
+        printUsage();
+        process.exit();
+      }
+      return true;
+
+    case "":
+    case undefined:
+    case "check-consistency":
```
Comment (folex) [inline ci.cjs:30]: what does that do?

File: ci.cjs [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  );
+}
+
+let postfix;
+const mode = process.argv[2];
+
+function validateArgs() {
+  switch (mode) {
+    case "get-version":
+      return true;
```
Comment (folex) [inline ci.cjs:18]: why doesn't it print version?

File: ci.cjs [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+      }
+      return true;
+
+    case "":
+    case undefined:
+    case "check-consistency":
+      return true;
+
+    default:
+      return false;
```
Comment (folex) [inline ci.cjs:34]: Maybe print `mode` with `console.err`, to report an error? and exit with 1, I guess

File: ci.cjs [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+function validateArgs() {
+  switch (mode) {
+    case "get-version":
+      return true;
+
+    case "bump-version":
+      postfix = process.argv[3];
+      if (!postfix) {
+        printUsage();
+        process.exit();
```
Comment (folex) [inline ci.cjs:24]: exit(1)?

File: ci.cjs (hunk 2 above)
Comment (nahsi) [reply ci.cjs:30]: I don't really know. It is a copy paste script that was written long time ago by Pavel. I copy paste it and change it a little bit.

I plan to this logic to to github-actions that publishes pnpm snapshots in future...

File: ci.cjs [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+async function processPackageJsons(allPackageJsons, versionsMap, fn) {
+  await Promise.all(allPackageJsons.map((x) => fn(x, versionsMap)));
+}
+
+async function run() {
+  if (!validateArgs()) {
+    printUsage();
+    process.exit(0);
```
Comment (folex) [inline ci.cjs:142]:
```suggestion
    process.exit(1);
```

File: ci.cjs (hunk 3 above)
Comment (nahsi) [reply ci.cjs:18]: Same story. I don't know, I don't use it :)

File: ci.cjs (hunk 5 above)
Comment (folex) [inline ci.cjs:24]: Why do you do `process.exit` here, instead of `return false`?

Comment (nahsi) [reply ci.cjs:24]: Why? `--help` returns 0 usually

Comment (folex) [reply ci.cjs:24]: but it's not `--help` situation, it's "invalid arguments" situation, an error

File: ci.cjs (hunk 2 above)
Comment (nahsi) [reply ci.cjs:30]: I checks if arguments are valid. checkConsistency is called in line ~147

**Review by folex — APPROVED** [review]

**Pull request merged**